apiserver/src/main/java/org/dependencytrack/event/kafka/processor/ProcessedVulnerabilityScanResultProcessor.java

```diff
@@ -261,18 +263,23 @@ private static List<WorkflowState> updateWorkflowStates(final Handle jdbiHandle,
         final var workflowDao = jdbiHandle.attach(WorkflowDao.class);
         final Set<UUID> workflowTokensWithBomProcessed =
                 workflowDao.getTokensByStepAndStateAndTokenAnyOf(WorkflowStep.BOM_PROCESSING, WorkflowStatus.COMPLETED, workflowTokens);
-        if (workflowTokensWithBomProcessed.isEmpty()) {
-            LOGGER.debug("None of the possible %d workflows have %s steps with status %s"
-                    .formatted(workflowTokens.size(), WorkflowStep.BOM_PROCESSING, WorkflowStatus.COMPLETED));
+        final Set<UUID> workflowTokensWithBomProcessFailed =
+                workflowDao.getTokensByStepAndStateAndTokenAnyOf(WorkflowStep.BOM_PROCESSING, WorkflowStatus.FAILED, workflowTokens);
```

Comment:
If the `BOM_PROCESSING` step failed, no vulnerability scan will have been performed in the first place.

Whether to send `BOM_PROCESSED` or `BOM_PROCESSING_FAILED` notifications here should depend on the outcome of the vulnerability scan (`completedVulnScan.getStatus()`).

Reply:
`workflowTokens` contain records for both completed and failed status of `completedVulnScan`. And `workflowTokensWithBomProcessFailed` would contain ones with `BOM_PROCESSING` step failed status (vuln-analysis not started yet).
It is adding notifications for both cases in the list and dispatching before further events. Do you mean dispatching should not be in `shouldDispatchBomProcessedNotification` if block ?

Reply:
I think you can do it here: 

https://github.com/DependencyTrack/hyades-apiserver/blob/eceeb6a44947bc765f6ea9605e6c904d3c639302/apiserver/src/main/java/org/dependencytrack/event/kafka/processor/ProcessedVulnerabilityScanResultProcessor.java#L270-L286

For tokens in `workflowTokensWithBomProcessed` where `completedVulnScan` has status `COMPLETED`, create `BOM_PROCESSED` notifications. If status is `FAILED`, create `BOM_PROCESSING_FAILED` instead.